### PR TITLE
fix(@aws-amplify/datastore): nextjs/swc broken build workaround

### PR DESCRIPTION
**Pull request by svidgen**
#### Description of changes

NextJS 13 enables SWC by default, which sometimes mangles the the build. This PR does two things:

1. Adds integ tests for sample apps that attempt to trigger bad nextjs builds
2. Adds a workaround for one known case, where SWC was over-aggressively dropping a variable declaration, resulting in an exception several lines later.

Note: Depends on https://github.com/aws-amplify/amplify-js-samples-staging/pull/494

#### Issue #, if available

#10732

#### Description of how you validated changes

Runtime failures were seen as in #10732 with NextJS@13 *prod* builds. The breakage was isolated by inserting dummy console logs with unique strings to help us determine where we were in the resulting minified build. By doing so, we could see that a variable declaration and later assignment were being eliminated entirely. Setting the var to a default value caused SWC to leave the declaration in and resolved runtime issues.

One of the newly added integ tests (the "basic" one) successfully triggers the error in CI on prod builds without the fix. The fix makes it pass in CI.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1354,6 +1354,57 @@ jobs:
...
+    parameters:
+      browser:
+        type: string
+    executor: js-test-executor
+    <<: *test_env_vars
+    working_directory: ~/amplify-js-samples-staging/samples/next/datastore/next-13-basic
+    steps:
+      - prepare_test_env
+      - integ_test_js:
+          test_name: 'DataStore - Nextjs basic'
```
Comment (jimblanc) [inline .circleci/config.yml:1367]: Nit: Should we add "SWC" to these test names just to make it a little more clear what the purpose is? E.g. `DataStore - Nextjs SWC basic`

File: .circleci/config.yml [hunk 2]
```diff
@@ -1910,7 +1961,33 @@ workflows:
...
+      - integ_next_datastore_13_basic:
+          requires:
+            - integ_setup
+            - build
+          filters:
+            <<: *releasable_branches
+          matrix:
+            parameters:
+              <<: *test_browsers
+      # - integ_next_datastore_13:
```
Comment (jimblanc) [inline .circleci/config.yml:1973]: Is this test a WIP? What's the difference between `integ_next_datastore_13_basic`?

**Review by jimblanc — DISMISSED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (svidgen) [reply .circleci/config.yml:1973]: The TS version of the test is failing for different nodejs compatibility reasons that I just didn't have time to dig into. Not sure if it's better to leave the uncommented here to re-add later or remove entirely.

**Review by jimblanc — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (svidgen) [reply .circleci/config.yml]: Renamed.

**Review by iartemiev — APPROVED** [review]
LGTM 👍🏻

**Review by dpilch — APPROVED** [review]

**Pull request merged**